cmake_modules/FindZSTD.cmake

```diff
@@ -24,7 +24,7 @@ endif()
 message (STATUS "ZSTD_HOME: ${ZSTD_HOME}")
 
 if (NOT ZSTD_STATIC_LIB_NAME)
-  set (ZSTD_STATIC_LIB_NAME ${CMAKE_STATIC_LIBRARY_PREFIX}zstd)
+  set (ZSTD_STATIC_LIB_NAME ${CMAKE_STATIC_LIBRARY_PREFIX}zstd${CMAKE_STATIC_LIBRARY_SUFFIX})
```

Comment:
I understand the use case, but this is used at line 35, `find_library (ZSTD_LIBRARIES NAMES ${ZSTD_STATIC_LIB_NAME} HINTS`. Usually, `find_library` doesn't require suffix, does it?

Reply:
It doesn't, however, depending on the order of items in the value of `CMAKE_FIND_LIBRARY_SUFFIXES`, it can pick up the shared library here (`libzstd.so`) and then fail later, while linking statically. IMO we should close this PR and open an issue to support linking third party dependency dynamically and statically, both.

Reply:
Could you open a PR with that approach? We can close this after that PR is merged.

Reply:
Yeah, we really need to clean up the CMakefiles. At this point, my original goal of using static libraries has become much less relevant. We should probably just default to using dynamic linking everywhere.

Reply:
Thanks, @omalley . Could you try to make a PR, @nehaljwani ?

Reply:
@omalley @dongjoon-hyun I've opened another PR: https://github.com/apache/orc/pull/541